app/controllers/fast_controller.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+# A proxy to Settings.autocomplete_lookup.url to get OCLC's FAST data for typeahead
+class FastController < ApplicationController
+  skip_verify_authorized only: %i[show]
+
+  include Dry::Monads[:result]
+
+  def show
+    result = lookups(params.require(:q))
+    @suggestions = result.value_or([])
+    if result.success?
+      return render status: :no_content if @suggestions.empty?
+
+      render status: :ok, layout: false
+    else
+      logger.warn(result.failure)
+      render status: :internal_server_error, html: ''
```

Comment:
Are we sure this is what we want to do in this case? Should be we treat as if no content and HB alert?

Reply:
@justinlittman It's what H2 does 🤷🏻‍♂️

Reply:
Oh, and I think we're logging the OCLC errors to HB in `lookups` below. So I think we're good.